rdbms/src/main/java/io/spine/server/storage/jdbc/query/DbIterator.java

```diff
@@ -237,41 +235,77 @@ protected R readResult() throws SQLException {
+     *         the type of the storage records in the first column
+     * @param <R2>
+     *         the type of the storage records in the second column
      */
-    private static class RecordWithIdIterator<I> extends DbIterator<EntityRecordWithId<I>> {
+    private static class DoubleColumnIterator<R1, R2>
+            extends DbIterator<DoubleColumnRecord<R1, R2>> {
 
-        private final ColumnReader<I> idColumnReader;
-        private final ColumnReader<EntityRecord> recordColumnReader;
+        private final ColumnReader<R1> firstColumnReader;
+        private final ColumnReader<R2> secondColumnReader;
 
         /**
-         * Creates a new instance of the {@code RecordWithIdIterator}.
+         * Creates a new instance of the {@code DoubleColumnIterator}.
          *
          * @param resultSet
          *         the SQL query results to iterate over
-         * @param idColumnReader
-         *         the reader of the ID column values
-         * @param recordColumnReader
-         *         the reader of the column storing entity records
+         * @param firstColumnReader
+         *         the reader for the values of the first column
+         * @param secondColumnReader
+         *         the reader for the values of the second column
          */
-        private RecordWithIdIterator(ResultSet resultSet,
-                                     ColumnReader<I> idColumnReader,
-                                     ColumnReader<EntityRecord> recordColumnReader) {
+        private DoubleColumnIterator(ResultSet resultSet,
+                                     ColumnReader<R1> firstColumnReader,
+                                     ColumnReader<R2> secondColumnReader) {
             super(resultSet);
-            this.idColumnReader = idColumnReader;
-            this.recordColumnReader = recordColumnReader;
+            this.firstColumnReader = firstColumnReader;
+            this.secondColumnReader = secondColumnReader;
         }
 
         @Override
-        protected EntityRecordWithId<I> readResult() throws SQLException {
-            I id = idColumnReader.readValue(resultSet());
-            EntityRecord record = recordColumnReader.readValue(resultSet());
-            EntityRecordWithId<I> result = EntityRecordWithId.of(id, record);
+        protected DoubleColumnRecord<R1, R2> readResult() throws SQLException {
+            R1 value1 = firstColumnReader.readValue(resultSet());
+            R2 value2 = secondColumnReader.readValue(resultSet());
+            DoubleColumnRecord<R1, R2> result = DoubleColumnRecord.of(value1, value2);
             return result;
         }
     }
+
+    /**
+     * A holder of a pair of column values for the simultaneous iteration over a {@link ResultSet}.
+     *
+     * @param <V1>
+     *         the type of the first column value
+     * @param <V2>
+     *         the type of the second column value
+     */
+    public static final class DoubleColumnRecord<V1, V2> {
```

Comment:
By convention, we use only single-symbol generic parameters. It's here and above.